STOR-45 refactor and fix last problems with StorageAPI

notefox opened the pull request:
refactored package structure for better navigation
added now second table usage for csv file-imported data 
used general refactor function of IntelliJ to automatically clean code
added extra Interfaces for Services, in case of future scaling
changed @Autowiring to use said Interfaces

Commit:
STOR-45 rename validator method to more fit the actual usage

Commit:
STOR-45 rename packages so they start with lower case

Commit:
STOR-45 general refactoring by using build-in "refactor code..." function of IntelliJ

Commit:
STOR-46 add interface for the StorageService

Commit:
STOR-46 add Validator interface

Commit:
STOR-46 rename packages

"custom_converter" to "customConverter"
"file_upload" to "fileUpload"

Commit:
STOR-46 refactor old StorageService interface to IFileStorageService

Commit:
STOR-46 add Interface for CSVConverter

Commit:
STOR-47 change Autowiring from ProductInformationValidator to IProductInformationValidator

Commit:
STOR-47 change Autowiring from StorageService to IStorageService

Commit:
STOR-47 refactor overuse of the Variable-name IFileStorageService after auto refactor

Commit:
STOR-47 add Interface to and refactor Autowired to use said Interface for CSVImport

Ktechen commented on src/main/java/com/butbetter/storage/model/Address.java:
```diff
@@ -150,14 +146,6 @@ public int hashCode() {
-				", postCode='" + postCode + '\'' +
-				", country='" + country + '\'' +
-				'}';
+		return "Address{" + "uuid=" + uuid + ", name='" + name + '\'' + ", companyName='" + companyName + '\'' + ", street='" + street + '\'' + ", city='" + city + '\'' + ", postCode='" + postCode + '\'' + ", country='" + country + '\'' + '}';
```
Don't remove autogen stuff with one liner

Ktechen commented on src/main/java/com/butbetter/storage/model/Address.java:
```diff
@@ -131,16 +131,12 @@ public void setCountry(String country) {
-				&& Objects.equals(city, address.city)
-				&& Objects.equals(postCode, address.postCode)
-				&& Objects.equals(country, address.country);
+		return Objects.equals(uuid, address.uuid) && Objects.equals(name, address.name) && Objects.equals(companyName, address.companyName) && Objects.equals(street, address.street) && Objects.equals(city, address.city) && Objects.equals(postCode, address.postCode) && Objects.equals(country, address.country);
```
Don't remove autogen stuff with one liner

Commit:
STOR-10 update response entity of handleFileUpload

Commit:
STOR-45 revert failed refactoring of Address and ProductInformation Objects

Commit:
STOR-12 update test file for csv importer test

Commit:
STOR-45 remove uuid from the need to be used in incoming csv file requests due to save issues

Commit:
STOR-45 removing uuid from the need to be used in incoming csv file requests due to save issues

Commit:
Merge remote-tracking branch 'origin/STOR-45-refactoring' into STOR-45-refactoring

Commit:
STOR-45 refactored general file-structure and table-usage in Postgres over Repositories

Commit:
STOR-25 remove old, now unused UUID Conversion method

Ktechen commented:
I am cool with that

Commit:
STOR-22 update storage.uml

Commit:
STOR-39 update uml diagram

Commit:
STOR-45 add better Error Handling in Controller on Exceptions

Commit:
STOR-45 IntelliJ refactor code

Ktechen commented on src/main/java/com/butbetter/storage/csvImport/controller/FileUploadController.java:
```diff
@@ -92,11 +80,50 @@ public FileUploadController(IFileStorageService fileStorageService) {
+	@ExceptionHandler(StorageFileNotFoundException.class)
+	public ResponseEntity<?> handleStorageFileNotFoundException(StorageFileNotFoundException e) {
+		logger.error(e.getMessage());
+		return ResponseEntity.internalServerError().body(e.getMessage());
```
I think best choice is bad request (400) because the internal server error is an unexpected condition.
Link: https://datatracker.ietf.org/doc/html/rfc7231#section-6.6.1

notefox replied:
Oh yeah, that's on me, 
the naming might be a bit off, will define another exception for internal server error and add it where it's needed

Commit:
STOR-45 (reaction to comment on pull request: https://github.com/But-Better/StorageApi/pull/11#discussion_r771930987)

add new StorageFileNotProcessableException for internal server error
StorageFileNotFound will now return a bad request

Commit:
update comments in converters to make code more clean
